packages/kokoas-client/src/pages/projContractV2/fields/ProfitRate.tsx

```diff
@@ -0,0 +1,83 @@
+import { Controller, useFormContext } from 'react-hook-form';
+import { TypeOfForm } from '../schema';
+import { InputAdornment, TextField } from '@mui/material';
+import { calculateAmount } from 'libs';
+
+export const ProfitRate = ({
+  disabled,
+}: {
+  disabled: boolean,
+}) => {
+  const { control, setValue, getValues } = useFormContext<TypeOfForm>();
+
+  return (
+    <Controller
+      name={'profitRate'}
+      control={control}
+      render={({
+        field: {
+          onBlur,
+          onChange,
+          ref,
+          value,
+          name,
+        },
+        fieldState: {
+          error,
+        },
+        
+      }) => {
+
+        return (
+          <TextField
+            label={'粗利率'}
+            inputRef={ref}
+            value={value}
+            name={name}
+            variant={'outlined'}
+            type='number'
+            onChange={(e) => {
+             
+
+              const profitRate = +e.target.value;
+             
+              if (!isNaN(profitRate)) {
+                onChange(profitRate);
+                const totalContractAmtAfterTax = getValues('totalContractAmtAfterTax');
+                const taxRate = getValues('taxRate');
+
+                const {
+                  amountBeforeTax,
+                  profit,
+                } = calculateAmount({
+                  taxRate: taxRate,
+                  amountAfterTax: totalContractAmtAfterTax,
+                  profitRate: profitRate / 100,
+                });
+
+                setValue('totalContractAmtBeforeTax', amountBeforeTax || 0);
+                setValue('totalProfit', profit || 0);
+              }
```

Comment:
```suggestion
                const {
                  amountBeforeTax,
                  profit,                  
                  costPrice,
                } = calculateAmount({
                  taxRate: taxRate,
                  amountAfterTax: totalContractAmtAfterTax,
                  profitRate: profitRate / 100,
                });

                setValue('totalContractAmtBeforeTax', amountBeforeTax || 0);
                setValue('totalProfit', profit || 0);
                setValue('costPrice', costPrice || 0);
              }
```

粗利率が変更された際、原価が更新されていませんでした。
こちらで動作確認した際には、上記変更でうまくいきそうでしたが、
react-hook-formに対してまだ経験が浅い為、処理の確認をお願いします。

Reply:
助かりました！そのまま適用しました。ありがとうございます。